File: MyFinance.Frontend/src/types/ReajusteAporteModo.ts
/**
 * Como o aporte mensal recorrente é reajustado a cada 12 meses simulados.
 * Corresponde ao enum ReajusteAporteModo.cs no backend (serializado como número).
 */
export const ReajusteAporteModo = {
    Nenhum: 1,
    PercentualFixo: 2,
    Ipca: 3,
} as const;

export type ReajusteAporteModo = (typeof ReajusteAporteModo)[keyof typeof ReajusteAporteModo];

File: MyFinance.Domain/Enums/ReajusteAporteModo.cs
namespace MyFinance.Domain.Enums
{
    /// <summary>Como o aporte mensal recorrente é reajustado a cada 12 meses simulados.</summary>
    public enum ReajusteAporteModo
    {
        /// <summary>Aporte mensal constante, sem reajuste.</summary>
        Nenhum = 1,

        /// <summary>Reajuste por um percentual fixo ao ano, informado pelo usuário.</summary>
        PercentualFixo = 2,

        /// <summary>Reajuste pelo IPCA anual real, buscado via Banco Central.</summary>
        Ipca = 3
    }
}
